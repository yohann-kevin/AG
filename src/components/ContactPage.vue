<template>
  <div>
    <h1>
      Bient√¥t disponible
    </h1>
  </div>
</template>

<script>
  export default {
    name: 'ContactPage',

    data: () => ({
    }),
  }
</script>
