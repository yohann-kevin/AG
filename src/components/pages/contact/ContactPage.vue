<template>
  <div class="contact-page">
    <h2>Contact</h2>
    <p>
      Une question ? Un problème ? Besoin de plus d'information ? N'hésitez pas
      à nous contacter via ce formulaire, nous nous engageons à vous répondre dans
      les plus brefs délais.
    </p>

    <div class="contact-container-alert">
      <v-alert
        dense
        text
        dismissible
        elevation="15"
        type="error"
        v-model="errorAlert"
        class="contact-alert"
      >
        Quelque chose c'est mal passer réessayer de nous contactez plus tard !
      </v-alert>
      <v-alert
        dense
        text
        dismissible
        elevation="15"
        type="success"
        v-model="successAlert"
        class="contact-alert"
      >
        Votre demande de contact à bien été envoyer !
      </v-alert>
    </div>

    <v-card
      elevation="8"
      class="contact-card"
    >
      <v-form
        class="contact-form"
        ref="contact-form"
        v-model="valid"
      >
        <v-text-field
          v-model="email"
          color="black"
          :rules="emailRules"
          type="email"
          label="Courriel"
          class="contact-input"
          required
        />

        <v-text-field
          v-model="object"
          color="black"
          label="Objet de votre demande"
          class="contact-input"
          required
        />

        <v-textarea
          v-model="content"
          color="black"
          label="Votre message"
          class="contact-input"
          filled
          required
        />
      </v-form>

      <div class="contact-btn">
        <v-btn
          text
          :disabled="!valid"
          @click="sendEmail"
        >
          Envoyer
        </v-btn>
        <v-btn
          text
          @click="resetForm"
        >
          Annuler
        </v-btn>
      </div>
    </v-card>
  </div>
</template>

<script>
export default {
  data: () => ({
    valid: false,
    successAlert: false,
    errorAlert: false,
    email: "",
    emailRules: [
      (v) =>
        /^[\w-\\.]+@([\w-]+\.)+[\w-]{2,4}$/.test(v) || "Adresse email invalide",
    ],
    object: "",
    content: ""
  }),
  methods: {
    resetForm() {
      this.$refs['contact-form'].reset();
    },
    sendEmail() {
      const data = {
        sender: this.email,
        object: this.object,
        content: this.content
      };

      const config = {
        method: 'post',
         // eslint-disable-next-line no-undef
         url: process.env.VITE_APP_API_URL + 'contacts',
        headers: { 
          'Content-Type': 'application/json'
        },
        data : data
      };

      this.$axios(config).then(response => {
        if(response.status === 201) {
          this.successAlert = true;
          this.$refs['contact-form'].reset();
        }
      }).catch(error => {
        this.errorAlert = true;
        console.log(error);
      });

    }
  }
};
</script>

<style scoped>
.contact-page {
  width: 100%;
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
}

.contact-page > h2 {
  width: 100%;
  text-align: center;
  margin: 20px;
}

.contact-page > p {
  width: 80%;
  text-align: center;
  margin: 15px;
}

.contact-container-alert {
  width: 60%;
  margin: 15px;
}

.contact-alert {
  width: 100%;
}

.contact-card {
  width: 60%;
  margin: 15px;
  padding: 20px;
}

.contact-btn {
  width: 100%;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}
</style>
