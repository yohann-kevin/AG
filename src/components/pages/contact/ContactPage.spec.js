import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import ContactPage from './ContactPage.vue';

describe('ResetPassword', () => {
  const initWrapper = () => {
    const localVue = createLocalVue();
    const vuetify = new Vuetify();
    return shallowMount(ContactPage, {
      localVue,
      vuetify
    });
  };

  it('should test component render correctly', () => {
    const wrapper = initWrapper();

    const title = wrapper.find('h2');
    expect(title.text()).toContain('Contact');
    expect(wrapper).toMatchSnapshot();
  });
});
