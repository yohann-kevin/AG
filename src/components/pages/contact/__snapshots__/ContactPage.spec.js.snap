// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword should test component render correctly 1`] = `
<div class="contact-page">
  <h2>Contact</h2>
  <p>
    Vous souhaitez obtenir plus d'informations sur nos prestations ?
    Décrivez-nous votre projet à l’aide du formulaire ci-dessous.
    Nous nous engageons à vous répondre dans les plus brefs délais
    pour vous faire une proposition d’accompagnement.
  </p>
  <v-card elevation="8" class="contact-card">
    <v-form class="contact-form">
      <v-text-field color="black" rules="v => /^[\\w-\\\\.]+@([\\w-]+\\.)+[\\w-]{2,4}$/.test(v) || &quot;Adresse email invalide&quot;" type="email" label="Courriel" prepend-icon="mdi-email" required="" class="contact-input"></v-text-field>
      <v-text-field label="Votre numéro de téléphone" rules="v => /^(?:(?:\\+|00)33|0)\\s*[1-9](?:[\\s.-]*\\d{2}){4}$/.test(v) || &quot;Numéro de téléphone invalide&quot;" color="black" prepend-icon="mdi-phone" class="contact-input"></v-text-field>
      <v-text-field color="black" label="Nom de votre entreprise" prepend-icon="mdi-home-city" class="contact-input"></v-text-field>
      <v-text-field color="black" label="Votre secteur d'activité" prepend-icon="mdi-briefcase" class="contact-input"></v-text-field>
      <v-text-field color="black" label="Objet de votre demande" required="" prepend-icon="mdi-email-outline" class="contact-input"></v-text-field>
      <v-select items="[object Object],[object Object],[object Object],[object Object]" label="Votre besoin" color="black" item-color="black" multiple="" chips="" required="" prepend-icon="mdi-magnify" class="contact-input"></v-select>
      <v-textarea color="black" label="Votre message" filled="" required="" prepend-icon="mdi-message" class="contact-input"></v-textarea>
    </v-form>
    <div class="contact-btn">
      <v-btn text="">
        Envoyer
      </v-btn>
      <v-btn text="">
        Annuler
      </v-btn>
    </div>
  </v-card>
  <div class="contact-container-alert">
    <v-alert dense="" text="" dismissible="" elevation="15" type="error" class="contact-alert">
      Quelque chose c'est mal passer réessayer de nous contactez plus tard !
    </v-alert>
    <v-alert dense="" text="" dismissible="" elevation="15" type="error" class="contact-alert">
      Votre email, votre besoin, votre message et l'objet de celui-ci sont requis pour envoyer une demande de contact !
    </v-alert>
    <v-alert dense="" text="" dismissible="" elevation="15" type="success" class="contact-alert">
      Votre demande de contact à bien été envoyer !
    </v-alert>
    <!---->
  </div>
</div>
`;
